src/api/define.js

```diff
@@ -1,146 +1,36 @@
-      // attributeChangedCallback.
-      if (observedAttributes.indexOf(linkedAttr) === -1) {
-        observedAttributes.push(linkedAttr);
-      }
-    }
-  });
-
-  // Merge observed attributes.
-  Object.defineProperty(Ctor, 'observedAttributes', {
-    configurable: true,
-    enumerable: true,
-    get () {
-      return observedAttributes;
-    }
-  });
-}
-
-function createInitProps (Ctor) {
-  const props = ensurePropertyDefinitions(Ctor);
-
-  return (elem) => {
-    if (!props) {
-      return;
-    }
-
-    keys(props).forEach((name) => {
-      const prop = props[name];
-      prop.created(elem);
-
-      // We check here before defining to see if the prop was specified prior
-      // to upgrading.
-      const hasPropBeforeUpgrading = name in elem;
-
-      // This is saved prior to defining so that we can set it after it it was
-      // defined prior to upgrading. We don't want to invoke the getter if we
-      // don't need to, so we only get the value if we need to re-sync.
-      const valueBeforeUpgrading = hasPropBeforeUpgrading && elem[name];
-
-      // https://bugs.webkit.org/show_bug.cgi?id=49739
-      //
-      // When Webkit fixes that bug so that native property accessors can be
-      // retrieved, we can move defining the property to the prototype and away
-      // from having to do if for every instance as all other browsers support
-      // this.
-      Object.defineProperty(elem, name, prop);
-
-      // We re-set the prop if it was specified prior to upgrading because we
-      // need to ensure set() is triggered both in polyfilled environments and
-      // in native where the definition may be registerd after elements it
-      // represents have already been created.
-      if (hasPropBeforeUpgrading) {
-        elem[name] = valueBeforeUpgrading;
-      }
-    });
-  };
-}
-
-function generateUniqueName (name) {
-  // http://stackoverflow.com/questions/105034/create-guid-uuid-in-javascript/2117523#2117523
-  const rand = 'xxxxxxxx'.replace(/[xy]/g, (c) => {
-    const r = Math.random() * 16 | 0;
-    const v = c === 'x' ? r : (r & 0x3 | 0x8); // eslint-disable-line no-mixed-operators
-    return v.toString(16);
-  });
-
-  return `${name}-${rand}`;
-}
-
-function prepareForRegistration (name, Ctor) {
-  Ctor[$name] = name;
-  Ctor[$props] = createInitProps(Ctor);
-}
-
 export default function (name, opts) {
-  if (opts === undefined) {
-    throw new Error(`You have to define options to register a component ${name}`);
-  }
-  const Ctor = typeof opts === 'object' ? Component.extend(opts) : opts;
   const { customElements } = root;
-  formatLinkedAttributes(Ctor);
 
   if (!customElements) {
     throw new Error('Skate requires native custom element support or a polyfill.');
   }
 
-  let uniqueName = name;
-  if (customElements.get(name)) {
-    uniqueName = generateUniqueName(name);
+  // Support passing an anonymous definition.
+  if (!opts) {
```

Comment:
To check, just in case: if I pass just a name here (like an idiot, yeah), it will fail. Should it do that? Or maybe some sort of warning would be better?
```
define('x-weird-custom-element');
```